pkg/acme/client/interfaces.go

```diff
@@ -40,4 +44,35 @@ type Interface interface {
 	UpdateReg(ctx context.Context, a *acme.Account) (*acme.Account, error)
 }
 
-var _ Interface = &acme.Client{}
+var _ Interface = &acme.Client{
+	// inspired by acme/http.go
+	RetryBackoff: func(n int, r *http.Request, res *http.Response) time.Duration {
+		var jitter time.Duration
+		if x, err := rand.Int(rand.Reader, big.NewInt(1000)); err == nil {
+			// Set the minimum to 1ms to avoid a case where
+			// an invalid Retry-After value is parsed into 0 below,
+			// resulting in the 0 returned value which would unintentionally
+			// stop the retries.
+			jitter = (1 + time.Duration(x.Int64())) * time.Millisecond
+		}
+		if _, ok := res.Header["Retry-After"]; ok {
+			// if Retry-After is set we should
+			// error and let the cert-manager logic retry instead
+			return -1
+		}
+
+		// classic backoff here in case we got no reply
+		// eg. flakes
```

Comment:
I'd suggest never retrying and always relying on the cert-manager retry logic.
All temporary errors would then be surfaced and logged as events linked to the order.
This retry function could be simplified to a single line.
What would be the downside of that?

And does the code below retry forever? 
What's the point if we elsewhere apply a timeout context of 10seconds.